actionpack/lib/action_controller/railtie.rb

```diff
@@ -24,7 +24,7 @@ class Railtie < Rails::Railtie #:nodoc:
     initializer "action_controller.parameters_config" do |app|
       options = app.config.action_controller
 
-      ActiveSupport.on_load(:action_controller) do
+      ActiveSupport.on_load(:action_controller_base) do
```

Comment:
No, it is executed twice. One for `ActionController::Base` and other for `ActionController::API`. This is why changing to `on_load(:action_controller_base)` fix the problem, because it only run for `ActionController::Base`